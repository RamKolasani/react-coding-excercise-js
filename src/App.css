*{
  padding:0px;
  margin:0px;

}
body{
  background-color: #c9cace;
  font-family: Arial, Helvetica, sans-serif;
  font-stretch: expanded;
  word-spacing: 2px;
  line-height: 1.5em;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  /* display: flex;
  flex-direction: row;
  flex-wrap:wrap;
  justify-content: center; */
  font-size: calc(10px + 2vmin);
  color: white;
  margin:auto;
}

.App-link {
  color: #c0cbce;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

ul{
  list-style-type: none;
}

.display-items{
  background-color: white;
  padding:10px;
  border-radius: 0px;
  margin: 5px;
  height: 200px;
  width:600px;
  display: flex;
  flex-direction: row;  
}

.display-items:hover{
  background-color: #FFA979;
  cursor: pointer;
}

.item-image{
  height: 200px;
  width: 200px;
  min-width: 200px;
  border: 1px solid grey;
  border-radius: 5px;
  
}

.item-details{
  margin-left: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.items-list{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.price{
  color: green;
  font-size:16px;
}

.not-available{
  color: 1px solid red;
}

.item-name{
  color: blue;
  text-align: center;
  font-stretch: extra-expanded;
  font-family: sans-serif;
  padding-top:5px;
}

.view-item{
  background-color:white;
  padding:10px;
  margin:60px;
  min-width: 600px;
  min-height: 600px;
}

.items-list a{
  text-decoration: none;
  color: black;
}

.view-item img{
  width: 400px;
  height: 400px;
  border: 1px solid grey;
  border-radius: 5px;
}



#nav-bar-button{
  padding : 10px;
  background-color: white;
  color: grey;
  border-radius: 5px;
  border: 1px solid grey;
  width: 180px;
  margin: 5px;
}

#nav-bar-button:hover{
  background-color: darkgrey;
  color:black;
  cursor: pointer;
}


#content{
  display: flex;
  flex-direction: row;
  justify-content: center;
  line-height: 2em;
}

#content img{
  margin-right: 50px;
  
}

#save-button{
  padding :10px;
  margin :10px 10px 0px 0px;
  background-color: white;
  color:black;
  width: 100px;
  border-radius: 5px;
}

#cancel-button{
  padding :10px;
  margin :10px 0px 0px 0px;
  background-color: white;
  color:black;
  border-radius: 5px;
  width: 100px;
}

#save-button:hover{
  cursor: pointer;
  background-color: lightseagreen;
}

#cancel-button:hover{
  cursor: pointer;
  background-color:lightpink;
}

.edit-pen:hover{
  cursor:pointer;
}

#view-item-details p{
  max-width: 400px;
  overflow: hidden;
}


.availability-selector {
  padding: 5px;
  height: 30px;
  width: 200px;
}

option span{
  padding: 5px;
  height: 30px;
  width: 200px;
}

.price-text{
  height: 20px;
  width: 80px;
  padding: 5px;
}

