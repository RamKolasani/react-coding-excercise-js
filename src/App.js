import React from 'react';

import './App.css';
import menuData from './menu-data-backup.json';
import {BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Dashboard from './components/Dashboard';
import ViewItem from './components/ViewItem';




function App() {
  let MenuData= menuData;
  console.log(menuData);
  return (
    <Router>
      <Switch>
        <Route exact path='/' render={() => <Dashboard MenuData={MenuData} />} /> 
        <Route path='/view-item/:itemId' component={ViewItem} />
      </Switch>
    </Router>
  );
}

export default App;