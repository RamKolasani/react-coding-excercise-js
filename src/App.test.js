import React from 'react';
import App from './App';
import {shallow} from 'enzyme';

test('renders learn react link', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

test('should have 2 routes',()=>{
  const wrapper = shallow(<App />);
  expect(wrapper.find('Route')).toHaveLength(2);
})
