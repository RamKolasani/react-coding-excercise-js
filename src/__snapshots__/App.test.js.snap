// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/view-item/:itemId"
    />
  </Switch>
</BrowserRouter>
`;
